dan_layer/storage/src/consensus_models/block.rs

```diff
@@ -129,6 +132,8 @@ impl Block {
             foreign_indexes: sorted_foreign_indexes,
             stored_at: None,
             signature,
+            block_time: None,
+            timestamp: EpochTime::now().as_u64(),
```

Comment:
You cant use epoch time here since this will typically be different for each node, which will break things when timestamp is added to the signature. You need to add the timestamp field to the protobuf protposal message and set it from that.